packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -31,7 +33,7 @@ abstract class RunCommandBase extends FlutterCommand {
           'Note: this flag will be removed in favor of the --debug/--profile/--release flags.');
 
     argParser.addFlag('trace-startup',
-        negatable: true,
+        negatable: false,
```

Comment:
This is for the definition in `RunCommandBase`. For the normal run, we'll want `--trace-startup` disabled by default. It sounds like you may make it enabled by default for the drive command? We'd need to have two separate declarations then, one in `RunCommand` and one in `DriveCommand`.  It would be a little cleaner if the run command wasn't negatable, since `--no-trace-startup` doesn't make sense for that command.